### PR TITLE
Riva streaming s2t client

### PR DESCRIPTION
**Pull request by ydharavath**
Previously, when processing a folder of audio files using riva nmt streaming s2t client, the output in output.txt is overwritten with each new file processed. This results in only the last audio file's output being retained in the file.

So now the client is updated so that the files are not overwritten but appended and the output will now be shown in json format in the output file

### CODE REVIEW

File: riva/clients/nmt/riva_nmt_streaming_s2t_client.cc [hunk 1]
```diff
@@ -119,12 +124,12 @@ main(int argc, char** argv)
...
-
+  emptyFileBeforeRequest();
   if (argc > 1) {
     std::cout << gflags::ProgramUsage();
     return 1;
   }
-
+  
   bool flag_set = gflags::GetCommandLineFlagInfoOrDie("riva_uri").is_default;
   const char* riva_uri = getenv("RIVA_URI");
```
Comment (rmittal-github) [inline riva/clients/nmt/riva_nmt_streaming_s2t_client.cc]: all changes in this file are not seen on the MR. Seems only partial changes pushed.

File: riva/clients/nmt/riva_nmt_t2t_client.cc [hunk 2]
```diff
@@ -90,13 +90,31 @@ translateBatch(
   }
 }
 
+int countWords(const std::string& text) {
```
Comment (rmittal-github) [inline riva/clients/nmt/riva_nmt_t2t_client.cc]: these are unwanted changes from previous MR

File: riva/clients/nmt/riva_nmt_streaming_s2t_client.cc [hunk 3]
```diff
@@ -119,12 +124,12 @@ main(int argc, char** argv)
 
   std::signal(SIGINT, signal_handler);
   gflags::ParseCommandLineFlags(&argc, &argv, true);
-
+  emptyFileBeforeRequest();
```
Comment (rmittal-github) [inline riva/clients/nmt/riva_nmt_streaming_s2t_client.cc]: do we need this function? at the start of fresh run of client, new file should get created that would overwrite any previous content. We should not need to explicitly empty previously existing file.

File: riva/clients/nmt/riva_nmt_streaming_s2t_client.cc [hunk 4]
```diff
@@ -119,12 +124,12 @@ main(int argc, char** argv)
...
   if (argc > 1) {
     std::cout << gflags::ProgramUsage();
     return 1;
   }
-
+
```
Comment (rmittal-github) [inline riva/clients/nmt/riva_nmt_streaming_s2t_client.cc]: remove trailing spaces

File: riva/clients/nmt/streaming_s2t_client.cc [hunk 5]
```diff
@@ -267,25 +269,30 @@ StreamingS2TClient::ReceiveResponses(std::shared_ptr<S2TClientCall> call, bool a
...
+
+    call->latest_result_.partial_transcript = "";
+    call->latest_result_.partial_time_stamps.clear();
+
+    bool is_final = false;
     for (int r = 0; r < call->response.results_size(); ++r) {
       const auto& result = call->response.results(r);
 
+      if (result.is_final()) {
+        is_final = true;
```
Comment (rmittal-github) [inline riva/clients/nmt/streaming_s2t_client.cc]: why did we add this flag? it is not used anywhere.

File: riva/clients/nmt/streaming_s2t_client.cc [hunk 6]
```diff
@@ -256,6 +256,8 @@ StreamingS2TClient::PostProcessResults(std::shared_ptr<S2TClientCall> call, bool
     VLOG(1) << "Latency:" << lat << std::endl;
     latencies_.push_back(lat);
   }
+  std::ofstream result_file(nmt_text_file_, std::ios::app);
```
Comment (rmittal-github) [inline riva/clients/nmt/streaming_s2t_client.cc]: can we open/close file in the constructor/destructor? like how ASR client is doing. We can put the std::ofstream object as a class member variable.

File: riva/clients/nmt/riva_nmt_t2t_client.cc [hunk 7]
```diff
@@ -288,4 +288,4 @@ main(int argc, char** argv)
 
 
   // std::vector<std::thread> workers;
-}
+}
```
Comment (rmittal-github) [inline riva/clients/nmt/riva_nmt_t2t_client.cc]: looks left over change, no change should be needed in this file

File: riva/clients/nmt/streaming_s2t_client.cc (hunk 5 above)
Comment (ydharavath) [reply riva/clients/nmt/streaming_s2t_client.cc]: This flag is required in the appendResult function

File: riva/clients/nmt/streaming_s2t_client.cc [hunk 8]
```diff
@@ -68,7 +68,7 @@ StreamingS2TClient::StreamingS2TClient(
   num_active_streams_.store(0);
   num_streams_finished_.store(0);
   thread_pool_.reset(new ThreadPool(4 * num_parallel_requests));
-
+  output_file_.open(nmt_text_file);
```
Comment (rmittal-github) [inline riva/clients/nmt/streaming_s2t_client.cc:74]: missing close call, can you add that in the destructor?

Comment (ydharavath) [reply riva/clients/nmt/streaming_s2t_client.cc:74]: I added it to the destructor

File: riva/clients/nmt/riva_nmt_t2t_client.cc [hunk 9]
```diff
@@ -90,7 +90,9 @@ translateBatch(
   }
 }
 
+<<<<<<< HEAD
```
Comment (rmittal-github) [inline riva/clients/nmt/riva_nmt_t2t_client.cc]: unwanted changes again appearing on the MR

File: riva/clients/nmt/streaming_s2t_client.cc [hunk 10]
```diff
@@ -269,9 +274,10 @@ StreamingS2TClient::ReceiveResponses(std::shared_ptr<S2TClientCall> call, bool a
     gotoxy(0, 5);
   }
 
-  std::ofstream result_file(nmt_text_file_);
   while (call->streamer->Read(&call->response)) {  // Returns false when no more to read.
     call->recv_times.push_back(std::chrono::steady_clock::now());
+
+
```
Comment (rmittal-github) [inline riva/clients/nmt/streaming_s2t_client.cc]: empty lines to be removed here

File: riva/clients/nmt/streaming_s2t_client.cc [hunk 11]
```diff
@@ -281,13 +287,11 @@ StreamingS2TClient::ReceiveResponses(std::shared_ptr<S2TClientCall> call, bool a
         gotoxy(0, 5);
       }
       VLOG(1) << "Result: " << result.DebugString();
-      std::cout << "translated text: \"" << result.alternatives(0).transcript() << "\""
-                << std::endl;
-      result_file << result.alternatives(0).transcript() << std::endl;
+      call->latest_result_.audio_processed = result.audio_processed();
+
```
Comment (rmittal-github) [inline riva/clients/nmt/streaming_s2t_client.cc]: this empty line can also be removed

**Review by rmittal-github — APPROVED** [review]

**Pull request merged**